src/DataProtection/DataProtection/src/Internal/KeyManagementOptionsSetup.cs

```diff
@@ -69,8 +70,12 @@ public void Configure(KeyManagementOptions options)
                 options.AuthenticatedEncryptorConfiguration = new AuthenticatedEncryptorConfiguration();
             }
 
-            options.AuthenticatedEncryptorFactories.Add(new CngGcmAuthenticatedEncryptorFactory(_loggerFactory));
-            options.AuthenticatedEncryptorFactories.Add(new CngCbcAuthenticatedEncryptorFactory(_loggerFactory));
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
@blowdart what do you think about this? I suspect the implementation no-ops in non-Windows which is why adding it was fine. We could suppress the warning here or mark the type as being supported on all platforms.

Reply:
I think you need to talk to @GrabYourPitchforks. Data Protection was designed whilst DI was limited and so has quirks. It looks fine to me, but he might known better here.